### PR TITLE
Shows prompt to create Skaffold configurations if any YAML file in the project becomes valid Skaffold file

### LINKED ISSUE #94: Detect new Skaffold file after it's created in the project
Currently, we detect Skaffold files only when we initially open a project. When a new file is created, we don't detect it and do not prompt to create Skaffold configurations automatically.

### PR DESCRIPTION
**Pull request by ivanporty**
Fixes #94.

Uses VFS manager and listener to track YAML file content and prompt to create Skaffold configs when new valid Skaffold YAML is created. This should complete automatic Skaffold presence detection in addition to file detection when a new project is opened. 

Immediately shows prompt to create Skaffold configurations if any YAML file in the project becomes valid Skaffold file and no Skaffold configurations exists.

### CODE REVIEW

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt [hunk 1]
```diff
@@ -66,17 +72,39 @@ class SkaffoldConfigurationDetector(val project: Project) : ProjectComponent {
...
+        // exist yet to show prompt for configurations once Skaffold file is created.
+        addVirtualFileListener(object : VirtualFileListener {
+            override fun contentsChanged(event: VirtualFileEvent) {
+                DumbService.getInstance(project).runWhenSmart {
+                    if (event.file.fileType is YAMLFileType &&
+                        SkaffoldFileService.instance.isSkaffoldFile(
+                            event.file
+                        ) && !hasExistingSkaffoldConfigurations()
+                    ) {
+                        // content changed to be a valid Skaffold file, no configurations, prompt
```
Comment (etanshaul) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt]: super nit: // content changed to be a valid Skaffold file, and no configurations exist, prompt

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt [hunk 2]
```diff
@@ -66,17 +72,39 @@ class SkaffoldConfigurationDetector(val project: Project) : ProjectComponent {
...
-                        .allConfigurationsList.filter { it is AbstractSkaffoldRunConfiguration }
-                if (skaffoldRunConfigList.isEmpty()) {
+                if (!hasExistingSkaffoldConfigurations()) {
                     // existing Skaffold config files, but no skaffold configurations, prompt
                     showPromptForSkaffoldConfigurations(project, skaffoldFiles[0])
                 }
             }
         }
+
+        // add VFS listener to ensure we track YAML file changes when no Skaffold configurations
```
Comment (etanshaul) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt]: perhaps move this comment to `addVirtualFileListener` or add another one there too

Comment (etanshaul) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt]: on second look. a different more generic comment on `addVirtualFileListener` seems more appropriate

Comment (ivanporty) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt]: done.

**Comment by ivanporty** [comment]
@etanshaul you were actually right about checking for file creation when file is copied from outside of the IDE, I'm adding this support as well, thanks for the idea!

**Comment by ivanporty** [comment]
done, PTAL

**Pull request merged**